coqpit/coqpit.py

```diff
@@ -278,6 +280,27 @@ def _deserialize(x: Any, field_type: Any) -> Any:
     raise ValueError(f" [!] '{type(x)}' value type of '{x}' does not match '{field_type}' field type.")
 
 
+# Recursive setattr (supports dotted attr names)
+def rsetattr(obj, attr, val):
+    def _setitem(obj, attr, val):
+        return operator.setitem(obj, int(attr), val)
+
+    pre, _, post = attr.rpartition('.')
+    setfunc = _setitem if pre.isnumeric() else setattr
+    return setfunc(rgetattr(obj, pre) if pre else obj, post, val)
+
+
+# Recursive getattr (supports dotted attr names)
+def rgetattr(obj, attr, *args):
+    def _getitem(obj, attr):
+        return operator.getitem(obj, int(attr), *args)
+
+    def _getattr(obj, attr):
+        getfunc = _getitem if attr.isnumeric() else getattr
+        return getfunc(obj, attr, *args)
+    return functools.reduce(_getattr, [obj] + attr.split('.'))
+
+
```

Comment:
I don't think this part correctly handles fields inherited from the parent class. 

You can try this. 
https://gist.github.com/erogol/23dc5cc3deb8fae714a98bbdcbdc370d

Reply:
That script passes for me with the changes in this PR.